nucliadb/tests/nucliadb/integration/test_ask.py

```diff
@@ -804,3 +818,44 @@ async def test_ask_on_resource_with_json_schema_automatic_prequeries(
     ask_response = SyncAskResponse.model_validate_json(resp.content)
     assert ask_response.prequeries is not None
     assert len(ask_response.prequeries) == 4
+
+
+async def test_all_rag_strategies_combinations(
+    nucliadb_reader: AsyncClient,
+    knowledgebox,
+    resources,
+):
+    rag_strategies = [
+        FullResourceStrategy(),
+        FieldExtensionStrategy(fields=["a/summary"]),
+        MetadataExtensionStrategy(types=list(MetadataExtensionType)),
+        HierarchyResourceStrategy(),
+        PreQueriesStrategy(queries=[PreQuery(request=FindRequest())]),
+    ]
+
+    def valid_combination(combination: list[RagStrategies]) -> bool:
+        try:
+            ChatRequest(query="foo", rag_strategies=combination)
+            return True
+        except ValueError:
+            return False
+
+    # Create all possible combinations of the list
+    valid_combinations = []
+    for i in range(1, len(rag_strategies) + 1):
+        for combination in combinations(rag_strategies, i):
+            if valid_combination(list(combination)):  # type: ignore
+                valid_combinations.append(list(combination))
+
+    assert len(valid_combinations) >= 19
```

Comment:
Why a `>=`, shouldn't be always the same number? Wouldn't be easier to count the invalid combinations?

Reply:
yes, it was just a temporary assert to make sure I was checking at least some combinations